beir/retrieval/models/onnx.py

```diff
@@ -77,16 +77,57 @@ def __tokenize_text(self, queries: List[str]) -> Tuple[np.ndarray, np.ndarray]:
         attention_mask = encoding['attention_mask']
         return input_ids, attention_mask
 
-    def __create_ort_input(self, queries: List[str]) -> Dict[str, List[np.ndarray]]:
-        ort_input_ids, ort_attention_mask = self.__tokenize_text(queries=queries)
+    def _create_ort_input(self, queries: List[str]) -> Dict[str, List[np.ndarray]]:
+        ort_input_ids, ort_attention_mask = self._tokenize_text(queries=queries)
         inputs = {
             'input_ids': list(ort_input_ids),
             'attention_mask': list(ort_attention_mask)
         }
         return inputs
 
-    def __batchify(self, queries: List[str], batch_size: int) -> List[List[str]]:
+    @staticmethod
+    def _batchify(queries: List[str], batch_size: int) -> List[List[str]]:
         batches = []
         for i in range(0, len(queries), batch_size):
             batches.append(queries[i: i+batch_size])
         return batches
+
+
+class OnnxBGE(OnnxBERT):
```

Comment:
It would be better to create an OnnxModel class that implements methods like batchify etc and then let OnnxBGE and OnnxBERT be children of this parent class. OnxxBGE being child of OnnxBERT works but is certainly not good design.

Reply:
Furthermore abstract methods can be declared so that any new model will have to implement them for the framework to work properly.

Reply:
I know, I did a quick and dirty solution just to add BGE embeddings to the framework. As you can see all the the functions were private and I just replaced them to protect just to avoid creating a third abstract class from where any other class will inherit or even just adding some ifs in the OnnxBERT class to enable query instruction.